lib/Doctrine/MongoDB/Aggregation/Expr.php

```diff
@@ -0,0 +1,890 @@
+     * The date argument can be any expression as long as it resolves to a date.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/dateToString/
+     * @param string $format
+     * @param mixed|self $expression
+     * @return self
+     */
+    public function dateToString($format, $expression)
+    {
+        return $this->operator('$dateToString', array($format, $expression));
+    }
+
+    /**
+     * Returns the day of the month for a date as a number between 1 and 31.
+     *
+     * The argument can be any expression as long as it resolves to a date.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/dayOfMonth/
+     * @param mixed|self $expression
+     * @return self
+     */
+    public function dayOfMonth($expression)
+    {
+        return $this->operator('$dayOfMonth', $expression);
+    }
+
+    /**
+     * Returns the day of the week for a date as a number between 1 (Sunday) and 7 (Saturday).
+     *
+     * The argument can be any expression as long as it resolves to a date.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/dayOfWeek/
+     * @param mixed|self $expression
+     * @return self
+     */
+    public function dayOfWeek($expression)
+    {
+        return $this->operator('$dayOfWeek', $expression);
+    }
+
+    /**
+     * Returns the day of the year for a date as a number between 1 and 366.
+     *
+     * The argument can be any expression as long as it resolves to a date.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/dayOfYear/
+     * @param mixed|self $expression
+     * @return self
+     */
+    public function dayOfYear($expression)
+    {
+        return $this->operator('$dayOfYear', $expression);
+    }
+
+    /**
+     * Divides one number by another and returns the result. The first argument is divided by the second argument.
+     *
+     * The arguments can be any valid expression as long as the resolve to numbers.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/divide/
+     * @param mixed|self $expression1
+     * @param mixed|self $expression2
+     * @return self
+     */
+    public function divide($expression1, $expression2)
+    {
+        return $this->operator('$divide', array($expression1, $expression2));
+    }
+
+    /**
+     * Compares two values and returns:
+     * true when the values are equivalent.
+     * false when the values are not equivalent.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/eq/
+     * @param mixed|self $expression1
+     * @param mixed|self $expression2
+     * @return self
+     */
+    public function eq($expression1, $expression2)
+    {
+        return $this->operator('$eq', array($expression1, $expression2));
+    }
+
+    /**
+     * Used to use an expression as field value. Can be any expression
```

Comment:
Just to confirm: this is the catch-all for handling custom expressions that we don't have builder methods for, correct?